src/main/java/ch/njol/skript/expressions/ExprBookPages.java

```diff
@@ -21,97 +21,165 @@
 import ch.njol.skript.Skript;
 import ch.njol.skript.aliases.Aliases;
 import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
 import org.bukkit.event.Event;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.BookMeta;
+import org.bukkit.inventory.meta.ItemMeta;
 import org.eclipse.jdt.annotation.Nullable;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 @Name("Book Pages")
 @Description("The pages of a book.")
-@Examples({"on book sign:",
-			"\tmessage \"Book Pages: %pages of event-item%\"",
-			"\tmessage \"Book Page 1: %page 1 of event-item%\""})
-@Since("2.2-dev31")
-public class ExprBookPages extends SimpleExpression<String> {
-	
+@Examples({
+	"on book sign:",
+	"\tmessage \"Book Pages: %pages of event-item%\"",
+	"\tmessage \"Book Page 1: %page 1 of event-item%\"",
+	"set page 1 of player's held item to \"Book writing\""
+})
+@Since("2.2-dev31, INSERT VERSION (changers)")
+public class ExprBookPages extends PropertyExpression<ItemType, String> {
+
 	static {
 		Skript.registerExpression(ExprBookPages.class, String.class, ExpressionType.PROPERTY,
-				"[all] [the] [book] (pages|content) of %itemtypes%",
-				"%itemtypes%'s [book] (pages|content)",
-				"[book] page %number% of %itemtypes%", 
-				"%itemtypes%'s [book] page %number%");
+			"[(all [[of] the]|the)] [book] (pages|content) of %itemtypes%",
+			"%itemtypes%'[s] [book] (pages|content)",
+			"[book] page %number% of %itemtypes%",
+			"%itemtypes%'[s] [book] page %number%");
 	}
-	
-	private static final ItemType bookItem = Aliases.javaItemType("book with text");
-	
-	@SuppressWarnings("null")
-	private Expression<ItemType> book;
+
 	@Nullable
 	private Expression<Number> page;
-	
-	@SuppressWarnings("null")
-	@Nullable
+
 	@Override
-	protected String[] get(Event e) {
-		ItemStack itemStack = book.getSingle(e).getRandom();
-		if (itemStack == null || !bookItem.isOfType(itemStack))
-			return null;
-		List<String> pages = ((BookMeta) itemStack.getItemMeta()).getPages();
-		if (page != null){
-			Number pageNumber = page.getSingle(e);
-			if (pageNumber == null){
-				return null;
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
```

Comment:
```suggestion
	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
```
Incorrect import, we only need ParseResult, import will need to be adjusted accordingly too.